bridges/darwinia-ethereum/bridge/src/service/guard.rs

```diff
@@ -165,33 +144,40 @@ impl GuardService {
             // high than last_confirmed(https://github.com/darwinia-network/bridger/issues/33),
             // and,
             // have not voted
-            if pending_block_number > last_confirmed
-                && !ethereum2darwinia.has_voted(&guard_account, voting_state)
-            {
-                match shadow.parcel(pending_block_number as usize).await {
-                    Ok(parcel_from_shadow) => {
-                        let parcel_from_shadow = parcel_from_shadow.try_into()?;
-                        let ex = if pending_parcel.is_same_as(&parcel_from_shadow) {
-                            Extrinsic::GuardVote(pending_block_number, true)
-                        } else {
-                            Extrinsic::GuardVote(pending_block_number, false)
-                        };
-                        extrinsics.push(ex);
-                    }
-                    Err(err) => {
-                        if let Some(BizError::BlankEthereumMmrRoot(block, msg)) =
-                            err.downcast_ref::<BizError>()
-                        {
-                            tracing::warn!(
-                                target: "darwinia-ethereum",
-                                "The parcel of ethereum block {} from Shadow service is blank, the err msg is {}",
-                                block,
-                                msg
-                            );
-                            return Ok(extrinsics);
-                        }
-                        return Err(err);
+            if last_confirmed <= pending_block_number {
+                continue;
```

Comment:
```
if pending_block_number > last_confirmed
                && !ethereum2darwinia.has_voted(&guard_account, voting_state)
```

Why continue and not vote here, different with the condition before change.

Reply:
The logic here is still the same, just the order has been adjusted.

before

```
for pending in pending_headers {
  if pending_block_number > last_confirmed
                  && !ethereum2darwinia.has_voted(&guard_account, voting_state)
              {
   // do something
  }
}
```

after

```
for pending in pending_headers {
  if last_confirmed <= pending_block_number {
    continue;
  }
  let real_account = client.account().real_account();
  let has_voted = voting_state.ayes.contains(real_account)
      || voting_state.nays.contains(real_account);
  if has_voted {
      continue;
  }

  // do something
}
```

https://github.com/darwinia-network/bridger/blob/c8c1a5babff6cc3d741fa4b26ce6d5b3e39c1ac6/bridges/darwinia-ethereum/client-darwinia/src/from_ethereum/api.rs#L305
https://github.com/darwinia-network/bridger/blob/c8c1a5babff6cc3d741fa4b26ce6d5b3e39c1ac6/frame/supports/support-ethereum/src/affirmation.rs#L41

Reply:
I thought again. This is indeed wrong